packages/grid/_modules_/grid/GridComponent.tsx

```diff
@@ -119,7 +120,7 @@ export const GridComponent = React.forwardRef<HTMLDivElement, GridComponentProps
                 )}
               >
                 <div ref={headerRef}>
-                  <components.Header {...componentBaseProps} {...props.componentsProps?.header} />
+                  <GridHeader />
```

Comment:
I don't think we agreed on removing the header slot?

Reply:
If you remove the header we are back to the same point and conversations where you remove the flexibility of moving components around and rendering them in other container...

Reply:
> If you remove the header we are back to the same point and conversation where you remove the flexibility of moving components around and rendering them in other container...

To which point I don't think is true. Hence the undocumentation path to prove it with having the option to quickly introduce it back (documenting)

Reply:
Ok my bad, I removed it because of 

> It seems to go in the right direction. I would also encourage we don't mention the header, at all, making it a private thing (removing it from the **docs**, components prop, etc.)

Ok my bad, I'll revert it and only remove it from the docs. I guess I got confused because of "component prop" 
got mentioned. I got left with the impression that the end goal here is to completely remove that component so in the end, we will have the following structure:

```
<DataGrid>
  <GridToolbar />
  <PreferencePanel />
  
  -- EVERYTHING ELSE --
</DataGrid>